src/poprox_recommender/components/joiners/interleave.py

```diff
@@ -0,0 +1,10 @@
+from poprox_concepts import ArticleSet
+
+
+class Interleave:
+    def __call__(self, candidates1: ArticleSet, candidates2: ArticleSet) -> ArticleSet:
+        articles = [None] * (len(candidates1.articles) + len(candidates2.articles))
+        articles[::2] = candidates1.articles
+        articles[1::2] = candidates2.articles
```

Comment:
If you're interested in going down the itertools route, itertools doesn't have interleave but more-itertools does.

Reply:
Of the indexing magic contained here, this is the one I think probably makes the most sense